mapping/react-leaflet.md

```diff
@@ -0,0 +1,433 @@
+
+Create the `onWardClick` function:
+```jsx
+function onWardClick(e) {
+        const layer = e.target
+
+        const layerFeature = (layer && layer.feature && layer.feature.properties) 
+                                ? layer.feature.properties 
+                                : null
+        
+        console.log(layerFeature)
+    }
+ ```
+
+To your `GeoJSON` instance, pass in the function as an argument to the `onEachFeature` props:
+```jsx
+<GeoJSON
+    key='ward-layer'
+    data={wardBorders}
+    style={fill} 
+    onEachFeature={eventHandlersOnEachFeature} />
+```
+
+Your `src/maps/wards.js` should look like this:
+```jsx
+import React, { useEffect, useState } from 'react'
+import BaseMap from './base'
+import { GeoJSON } from 'react-leaflet'
+
+function ChicagoWardMap() {
+    const [wardBorders, setWardBorders] = useState(null)
+
+    useEffect(() => {
+        // get the geojson
+        fetch('https://raw.githubusercontent.com/datamade/chicago-judicial-elections/master/wards/wards_2012.geojson')
+            .then((res) => res.json())
+            .then((geojson) => {
+                setWardBorders(geojson)
+            })
+
+    }, [setWardBorders])
+
+    const fill = {
+        fillColor: '#daf0ce', 
+        weight: 0.5,
+        opacity: 0.4,
+        color: '#666',
+        fillOpacity: 0.5
+    }
+    
+    function onWardClick(e) {
+        const layer = e.target
+
+        const layerFeature = (layer && layer.feature && layer.feature.properties) 
+                                ? layer.feature.properties 
+                                : null
+        
+        console.log(layerFeature)
+    }
+
+    function eventHandlersOnEachFeature(feature, layer) {
+        layer.on({
+          click: onWardClick
+        })
+    }
+
+    return (
+        <div className='map-viewer'>
+            <BaseMap center={[41.8781, -87.6298]} zoom={10}>
+                {wardBorders && <GeoJSON
+                                    key='ward-layer'
+                                    data={wardBorders}
+                                    style={fill} 
+                                    onEachFeature={eventHandlersOnEachFeature} />}
+            </BaseMap>
+        </div>
+    )
+}
+
+export default ChicagoWardMap
+```
+
+Now, whenever you click on a ward in your map, it should print some information in your console.
+
+### Show the ward's information in the UI
+Printing to the console isn't very useful. So, we need get that data and render it to the UI. Once we've done that, we'll refactor the component and "lift state up" to the parent component, so that we can keep the logic for a `ChicagoWardMap` separated from the logic of what you want on other parts of the UI.
+
+Create a new stateful object in the `ChicagoWardMap` component called `ward`:
+```jsx
+const [ward, setWard] = useState({})
```

Comment:
Is it `{}` or `null`?

Reply:
Also, can you say a bit more about what this code is doing? I can see someone (i.e., me) reading this and being like, but we didn't define a `setWard` method...